### DISCUSSION TITLE
适合在具有公网ip的内网服务器搭建吗？

### DISCUSSION THREAD

**Discussion by BBBoZi** [kkkgo/PaoPaoDNS#22] [问答和帮助]
可否请教一下直接作为递归合适吗？据说国内的根服务器污染挺多，不清楚是不是真的。

**Comment by kkkgo** [comment]
作为国内域名的递归解析是没有问题的。境外域名的话这个docker走的是加密查询，所以也没有问题。

**Comment by kkkgo** [comment]
实际上，在”验证你的递归DNS正常运行“的那个步骤，你就可以验证是否可行，如果你的网络环境真的有这么差（被劫持流量），验证就不会成功。

**Reply by BBBoZi** [reply]
也不是被劫持，只是表达如果直接成为递归服务器，直接查询根服务器，而据说国内根服务器的污染ip比较多，所以表达的意思就是成为递归是不是有可能返回ip被污染，可能并没有像查询dnspad递归服务器体验好，毕竟是公司运营，对其有优化。

**Reply by kkkgo** [reply]
顶级根服务仅仅是返回像com, net ,org ,cn这样的tld根，如果你域名放在dnspod上的话，递归最终还是会查询到dnspod上去的，并不存在被污染，在验证步骤我也提供了国内的递归域名测试服务`nslookup whoami.03k.org`，这是基于你的客户端查询IP返回的结果，如果你递归过程中被污染，那么他就不可能返回到正常IP。

**Reply by kkkgo** [reply]
实际上，DNS污染的方式并不是污染顶级根，而是抢答具体域名的A/AAAA记录。

**Reply by kkkgo** [reply]
> 也不是被劫持，只是表达如果直接成为递归服务器，直接查询根服务器，而据说国内根服务器的污染ip比较多，所以表达的意思就是成为递归是不是有可能返回ip被污染，可能并没有像查询dnspad递归服务器体验好，毕竟是公司运营，对其有优化。

如果根服务器遭到污染，”dnspod“这样的公司运营也不能幸免，以前就发生过这样的事情：
https://www.doit.com.cn/p/213242.html

**Reply by BBBoZi** [reply]
好的，非常感谢作者解答！
